### PR TITLE
Fix issue in ShareReplay publisher Switch to removing subscriptions after completion is sent

### PR DESCRIPTION
**Pull request by jabeattie**
Hey folks, you may remember my previous PR: https://github.com/CombineCommunity/CombineExt/pull/86

In this I tried to fix a memory leak that we'd identified, however, it seems like this solution caused a different problem that we've only uncovered when updating to 1.5.0.

I'm not 100% of what in our usage caused the issue - it is weirdly only presented in tests -but I've checked that this change both allows the tests in our consuming app to pass and works when the app is running for users.

I've removed the call to `cancel()` and instead clear the subscriptions after they are forwarded a completion. This seems to make sense to me logically, as once a subscriber has received a completion it can't do anything else.

I appreciate however that this is a change that had a problem for a new unknown change. If you aren't sure about this, I can instead revert my original PR. This would bring back the memory leak, but avoid the failing scenario we are seeing

### CODE REVIEW

**Comment by jabeattie** [comment]
@jasdev I know you reviewed the previous PR, would you be able to take a look at this?

**Comment by jasdev** [comment]
Seems reasonable to me after a quick check — any chance we can add a test to make sure this doesn’t accidentally regress down the line? 🙏🏽

**Comment by jabeattie** [comment]
Yeah absolutely. I'll try and write a failing test for the issue we experienced

File: Sources/Subjects/ReplaySubject.swift [hunk 1]
```diff
@@ -72,6 +72,10 @@ public final class ReplaySubject<Output, Failure: Error>: Subject {
         }
 
         subscriptions.forEach { $0.forwardCompletionToBuffer(completion) }
+
+        lock.lock()
+        defer { self.lock.unlock() }
+        self.subscriptions.removeAll()
```
Comment (freak4pc) [inline Sources/Subjects/ReplaySubject.swift:78]: so the point of this is to remove all of the subscriptions upon completion?

**Comment by mRs-** [comment]
I've created #124

**Comment by freak4pc** [comment]
Closed in favor of the now-merged #124. Thank you!

**Pull request closed without merging**